### ISSUE TITLE
Add backup and restore data functionality

### ISSUE THREAD

**Issue by let92** [lorenzovngl/FoodExpirationDates#64]
It would be great to see an option for backing up and restoring app database, e.g. Google Drive.

Hope it will be implemented in the future.

**Comment by lorenzovngl** [comment]
Agree. I'll take it into consideration, thanks for the suggestion!

**Comment by let92** [comment]
> Agree. I'll take it into consideration, thanks for the suggestion!

Thanks to you! I can't wait to see the feature implemented!

**Comment by BloodRaven0** [comment]
+1 this feature is essential if you have a partner (or room mate) who also does shopping in the household.

Ideally, it would be able to sync via nextcloud somehow, so people can self-host their own gorcery lists.

Using proprietary cloud services like google drive goes against the app's FOSS nature.

**Comment by Dtrieb** [comment]
Or sync with a list on Home Assistant.

**Comment by let92** [comment]
Any news?

**Comment by let92** [comment]
Any news? It would be very useful that this feature will be present in the app.

I hope you can develop this feature in the near future.

**Comment by nascentt** [comment]
Was thrilled to find this app, it's amazing. But the realization that either one person in the house has to manage all the food, or every had to scan and date everything independently (or repeatedly export and import lists) is frustrating.  Any way to syncing lists with people would be very much appreciated

**Comment by nascentt** [comment]
> Or sync with a list on Home Assistant.

I've just happened to setup up home assistant and have Our groceries integration which lets me check, add and remove things to my shopping list by voice and have it synched to my family's phones.

The very first thing I thought of was having this food expiration app linked to home assistant. Would be amazing to ask Nabi which foods are expiring today whilst starting to cook in the kitchen. 

I really hope someone is able to make this reality.

**Comment by BloodRaven0** [comment]
Same, I tried to use this app, but without sync functionality it's nearly unusable for me. Currently waiting for that implementayion to start using it again.

**Comment by let92** [comment]
Any news @lorenzovngl?

**Comment by let92** [comment]
@lorenzovngl?

**Comment by nascentt** [comment]
Doesn't sound like this being considered which is a big shame.  I'll point at #294 as being at least a workable compromise, where we could sync the list ourselves

**Comment by let92** [comment]
> Doesn't sound like this being considered which is a big shame. I'll point at [#294](https://github.com/lorenzovngl/FoodExpirationDates/issues/294) as being at least a workable compromise, where we could sync the list ourselves

Yeah, it would be nice that almost that possibility is available.

Hope in a developer reply...

**Comment by nascentt** [comment]
I ended up switching to Your Food - No Waste Inventory as that's got cloud sharing.  There is a 1 time fee (for only one of the people sharing) for the cloud sync functionality but it beats wasting tons of money throwing away food.
Hope one day this app adds such functionality.

**Comment by let92** [comment]
Hi @lorenzovngl,
thanks for the reply, the only thing that was needed was your response, not the day of release of the feature.

I understand this is a open source project and what it means, but also in open source there needs to be a dialogue between the project owner and those who have any feedback or request to make.

Since 29th September 2023, the day the feature request was opened, you've never responded to multiple requests. A single response like the one you've given now would have been enough, but not after two years.

Simply this: I hope the feature arrives, otherwise, it doesn't matter, I'll wait.

If it had been a C# project, I would definitely have contributed. Unfortunately, I haven't used Java since I studied it at university.

Thanks again for the reply.
